core/include/scipp/core/element/rebin.h

```diff
@@ -17,6 +17,74 @@
 
 namespace scipp::core::element {
 
+struct AscendingRebin {};
+struct DescendingRebin {};
+
+namespace {
+void rebin_error() {
+  throw std::runtime_error(
+      "Unknown type of rebin operation. "
+      "Possible branches are AscendingRebin and DescendingRebin.");
+}
+} // namespace
+
+template <class RebinType, class OldType, class NewType>
+bool compare_new_high_with_old_low(const NewType xn_high,
+                                   const OldType xo_low) {
+  if constexpr (std::is_same_v<RebinType, AscendingRebin>)
+    return xn_high <= xo_low;
+  else if constexpr (std::is_same_v<RebinType, DescendingRebin>)
+    return xn_high >= xo_low;
+  else
+    rebin_error();
+}
+
+template <class RebinType, class OldType, class NewType>
+bool compare_old_high_with_new_low(const OldType xo_high,
+                                   const NewType xn_low) {
+  if constexpr (std::is_same_v<RebinType, AscendingRebin>)
+    return xo_high <= xn_low;
+  else if constexpr (std::is_same_v<RebinType, DescendingRebin>)
+    return xo_high >= xn_low;
+  else
+    rebin_error();
+}
+
+template <class RebinType, class OldType, class NewType>
+double compute_delta(const NewType xn_high, const OldType xo_high,
+                     const NewType xn_low, const OldType xo_low) {
+  if constexpr (std::is_same_v<RebinType, AscendingRebin>)
+    return std::min<double>(xn_high, xo_high) -
+           std::max<double>(xn_low, xo_low);
+  else if constexpr (std::is_same_v<RebinType, DescendingRebin>)
+    return std::min<double>(xn_low, xo_low) -
+           std::max<double>(xn_high, xo_high);
+  else
+    rebin_error();
+}
```

Comment:
Still trying to think how to avoid this, maybe there is a way (given that none of the other helpers are required).

Reply:
Hah, `std::min` actually has an overload for this:
```cpp
template< class T, class Compare >
constexpr const T& min( const T& a, const T& b, Compare comp );
```
so with the changes I suggested for the other helpers (essentially just pass a `comp`), you can just use:
```cpp
const auto delta = std::min<double>(xn_high, xo_high, comp) - std::max<double>(xn_low, xo_low, comp);
```
(may need to throw an `std::abs` into the mix).